README.md

```diff
@@ -1,33 +1,174 @@
+await ApiClient.get('dealers/{dealerId}/listings/{listingId}', {
+  params: { dealerId: 123, listingId: 456 },
+})
+````
+
+### Authorization
+
+You can pass `accessToken` to the request options if you are talking to a protected API. The client will automatically
+set the header with a Bearer token.
+
+````typescript
+// fetch is called with the header { Authorization: `Bearer ${accessToken}` }
+ApiClient.post<{ id: number }, { name: string; listingIds: number[] }>(
+  "users/me/listing-comparisons",
+  { name: "test1", listingIds: [890163] },
+  { accessToken: authHeader.accessToken }
+)
+  .then((res) => {
+    // do something with the response
+  })
+  .catch((error) => {
+    // do something in case of an error
+  })
+````
+
+### Error handling
+
+The promise is rejected if any error occurs on API level. You can use promise chaining or `try/catch` to add error
+handling.
+
+````typescript
+ApiClient.post<LeasingCalculation, LeasingData>(
+  "listings/calculate-leasing",
+  {
+    downPayment: shouldWork ? 7300 : 17500,
+    duration: 48,
+    estimatedKmPerYear: shouldWork ? 10000 : 30000,
+    firstRegistrationDate: "2020-07-01",
+    residualValue: 12045,
+    price: 36500,
+  }
+)
+  .then((res) => {
+    // res is typed to LeasingCalculation
+  })
+  .catch((error: ResponseError) => {
+    // do any error handling you want
+  })
+
+
+try {
+  // res has the type LeasingCalculation
+  const res = await ApiClient.post<LeasingCalculation, LeasingData>(
+    "listings/calculate-leasing",
+    {
+      downPayment: shouldWork ? 7300 : 17500,
+      duration: 48,
+      estimatedKmPerYear: shouldWork ? 10000 : 30000,
+      firstRegistrationDate: "2020-07-01",
+      residualValue: 12045,
+      price: 36500,
+    }
+  )
+} catch (error: ResponseError) {
+  // do any error handling you want
+}
+````
+
+### Testing
+
+Create a file in the `__mocks__` directory called `@smg-automotive/api-client-pkg.ts` with the following content:
+
+````typescript
+export const ApiClient = {
+  get: jest.fn().mockReturnValue(Promise.resolve()),
+  post: jest.fn().mockReturnValue(Promise.resolve()),
+  put: jest.fn().mockReturnValue(Promise.resolve()),
+  delete: jest.fn().mockReturnValue(Promise.resolve()),
+}
+````
+
+By default, all the API calls will succeed. If you want to reject it or return a specific value you can do this as
+follows:
+
+````typescript
+jest.spyOn(ApiClient, "get").mockImplementation((path, options) => {
+  // do something with path and options (e.g. create a switch statement for the methods used)
+  return Promise.resolve([{ name: "bmw", key: "bmw" }])
+})
+
+jest.spyOn(ApiClient, "get").mockImplementation((path, options) => {
+  // do something with path and options (e.g. create a switch statement for the methods used)
+  return Promise.reject(
+    new ResponseError({ status: 422, statusText: "" }, "Some data from the response body")
+  )
+})
+
+// or using mock
+jest.mock("@smg-automotive/api-client-pkg", () => ({
```

Comment:
Wouldn't you need to require actual here?

Reply:
it works without it